### PR TITLE
Idemix Update

### PR DESCRIPTION
**Pull request by adecaro**
#### Type of change

- Improvement (improvement to code, performance, etc)

#### Description

This PR update the Idemix dependency to its latest version that is used by the Fabric Token SDK too.
Now, the SignerConfig struct also exports the RevocationHandle that is used by the Fabric Token SDK for audit purpose.
The way the revocation handle is encoded has changed to align to what the Idemix library expects.
The unit-tests have been updated.

#### Release Note

The changes to the encoding of the revocation handle should impact anyway given that Fabric is not currently using it.

### CODE REVIEW

**Comment by adecaro** [comment]
@denyeart @jkneubuh, please, could you help me here? It is not clear to me what's the issue. Thanks

**Comment by denyeart** [comment]
@adecaro Sorry for the delay, just getting back this week.

It looks like the issue is that an updated cobra got vendored in which adds a new `completion` option for the CA server and client commands, e.g.:
```
     Available Commands:
       completion  Generate the autocompletion script for the specified shell
       help        Help about any command
       init        Initialize the fabric-ca server
       start       Start the fabric-ca server
```

This results in the new `completion` option showing up in the generated docs at:
docs/source/clientcli.rst
docs/source/servercli.rst

The error indicates that the generated docs don't match the existing docs. To fix the issue, you would need to do one of two things:
1) run `make docs` to regenerate the docs and include the above files in your PR.
2) suppress the new `completion` option by following the instructions at https://www.reddit.com/r/golang/comments/11evntz/how_to_disable_cobra_cli_default_commands/

In my opinion, the completion feature is decent enough to keep. See description of it at https://blog.devgenius.io/shell-completion-with-cobra-and-go-c8368074d8f7.

**Comment by denyeart** [comment]
I've also opened a PR to clarify the confusing CI error message - https://github.com/hyperledger/fabric-ca/pull/365

**Comment by adecaro** [comment]
cool, thanks @denyeart. I'll run `make docs` and update the PR :)

**Comment by adecaro** [comment]
Hi @denyeart , anything missing that I need to address to get the PR merged? Thanks much :)

File: lib/client.go [hunk 1]
```diff
@@ -33,13 +23,23 @@ import (
 	idemixcred "github.com/hyperledger/fabric-ca/lib/client/credential/idemix"
 	x509cred "github.com/hyperledger/fabric-ca/lib/client/credential/x509"
 	cidemix "github.com/hyperledger/fabric-ca/lib/common/idemix"
+	idemix2 "github.com/hyperledger/fabric-ca/lib/server/idemix"
```
Comment (denyeart) [inline lib/client.go:26]: Might be good to have more expressive names than `idemix` and `idemix2`.

**Review by denyeart — COMMENTED** [review]
I don't completely understand the revocation handle changes, but I'll take your word for it.

So I've just added some nits from Go perspective.

Also, we typically prefer the PR author to squash their own commits into a single commit for a cleaner history upon merge.

File: lib/client.go (hunk 1 above)
Comment (adecaro) [reply lib/client.go:26]: `sidemix` should do the job

File: lib/server/idemix/enroll.go [hunk 2]
```diff
@@ -99,13 +99,19 @@ func (h *EnrollRequestHandler) HandleRequest() (*EnrollmentResponse, error) {
 	}
 
 	// Get revocation handle for the credential
-	rh, err := h.Issuer.RevocationAuthority().GetNewRevocationHandle()
+	rhOld, err := h.Issuer.RevocationAuthority().GetNewRevocationHandle()
```
[the remark this answers is not in the document]
Comment (adecaro) [reply lib/server/idemix/enroll.go:102]: Yeah, we don't need `rhOld` 👍

File: lib/client/credential/idemix/signerconfig.go [hunk 3]
```diff
@@ -20,8 +20,10 @@ type SignerConfig struct {
 	EnrollmentID string `protobuf:"bytes,5,opt,name=enrollment_id,json=enrollmentId" json:"enrollment_id,omitempty"`
 	// CRI contains a serialized Credential Revocation Information
 	CredentialRevocationInformation []byte `protobuf:"bytes,6,opt,name=credential_revocation_information,json=credentialRevocationInformation,proto3" json:"credential_revocation_information,omitempty"`
+	// RevocationHandle is the handle used to single out this credential and determine its revocation status
+	RevocationHandle string `protobuf:"bytes,7,opt,name=revocation_handle,json=revocationHandle,proto3" json:"revocation_handle,omitempty"`
 	// CurveID specifies the name of the Idemix curve to use, defaults to 'amcl.Fp256bn'
-	CurveID string `protobuf:"bytes,7,opt,name=curve_id,json=curveID" json:"curveID,omitempty"`
+	CurveID string `protobuf:"bytes,8,opt,name=curve_id,json=curveID" json:"curveID,omitempty"`
```
[the remark this answers is not in the document]
Comment (adecaro) [reply lib/client/credential/idemix/signerconfig.go:26]: makes sense

**Review by adecaro — COMMENTED** [review]
Comments addressed

**Review by denyeart — APPROVED** [review]

**Pull request merged**